gnuradio-runtime/lib/flowgraph.cc

```diff
@@ -73,9 +77,9 @@ void flowgraph::validate()
         std::vector<int> used_ports;
         int ninputs, noutputs;
 
-        if (FLOWGRAPH_DEBUG)
-            std::cout << "Validating block: " << (*p) << std::endl;
-
+#if FLOWGRAPH_DEBUG
```

Comment:
I would not make this change just to remove an include. `#if` has the drawback that it makes it more complex to do static analysis, breaks indentation, and risks breaking code that happens to not be active.

I would avoid preprocessor stuff unless it definitely can't compile otherwise. For C++17 we can use `if constexpr`.

Also elsewhere.

Reply:
I'm fine with that line of thinking. It's not a header, so the transitive include isn't hurting anyone. Let me revert that file

Reply:
(along with `hier_block2_detail.cc`)

reverted, @ThomasHabets.